common/lib/xmodule/xmodule/video_module/transcripts_utils.py

```diff
@@ -577,8 +577,14 @@ def get_video_transcript_content(edx_video_id, language_code):
     transcript = None
     edx_video_id = clean_video_id(edx_video_id)
     if edxval_api and edx_video_id:
-        transcript = edxval_api.get_video_transcript_data(edx_video_id, language_code)
-
+        try:
+            transcript = edxval_api.get_video_transcript_data(edx_video_id, language_code)
+        except Exception:
+            log.exception(
+                "Error getting transcript from edx-val id: {edx_video_id}: language code {language_code}"
+            )
+            content = '{"start": [],"end": [],"text": ["An error occured obtaining the transcript."]}'
+            transcript = dict(file_name='error-{language_code}.srt', content=content)
```

Comment:
What happens when this roundtrips back (export broken thing -> import again)?

The SRT file itself isn't supposed to be JSON encoded, right? Is this an error in the intermediate file format? Is there a way to get the source data file from `edx_val_api` in pre-processed state?

Reply:
Let me look into this... I think you are correct. This is therefore an error in the file format I added. Therefore, encoding `content` as srt should be the way to go. Considering edx-val's `get_video_transcript_data()` gives only .srt files back, it should be srt encoded.

Reply:
I'll do that now.

Reply:
Also, I might add in the video id to the file name.

Reply:
I have made it so it returns an .srt encoded file.